internal/controllers/database/init.go

```diff
@@ -62,9 +63,60 @@ func (r *Reconciler) setInitDatabaseCompleted(
 		Reason:  ReasonCompleted,
 		Message: message,
 	})
+	meta.SetStatusCondition(&database.Status.Conditions, metav1.Condition{
+		Type:   CreateDatabaseOperationCondition,
+		Status: metav1.ConditionTrue,
+		Reason: ReasonCompleted,
+	})
 	return r.updateStatus(ctx, database, StatusUpdateRequeueDelay)
 }
 
+func (r *Reconciler) checkCreateTenantOperation(
+	ctx context.Context,
+	database *resources.DatabaseBuilder,
+	tenant *cms.Tenant,
+	ydbOptions ydb.Option,
+) (bool, ctrl.Result, error) {
+	condition := meta.FindStatusCondition(database.Status.Conditions, CreateDatabaseOperationCondition)
+	if condition == nil || len(condition.Message) == 0 {
+		meta.RemoveStatusCondition(&database.Status.Conditions, CreateDatabaseOperationCondition)
+		return r.updateStatus(ctx, database, DatabaseInitializationRequeueDelay)
+	}
+	operationID := condition.Message
+	finished, err := tenant.CheckCreateOperation(ctx, operationID, ydbOptions)
+	if err != nil {
+		r.Recorder.Event(
+			database,
+			corev1.EventTypeWarning,
+			"InitializingFailed",
+			fmt.Sprintf("Error creating tenant %s: %s", tenant.Path, err),
+		)
+	}
+	if !finished {
+		if err == nil {
+			r.Recorder.Event(
+				database,
+				corev1.EventTypeWarning,
+				"Pending",
+				fmt.Sprintf("Tenant creation operation is not completed, operationID: %s", operationID),
+			)
+		}
+		return Stop, ctrl.Result{RequeueAfter: DatabaseInitializationRequeueDelay}, nil
+	} else if err != nil {
+		meta.RemoveStatusCondition(&database.Status.Conditions, CreateDatabaseOperationCondition)
+		return r.updateStatus(ctx, database, DatabaseInitializationRequeueDelay)
+	}
```

Comment:
```suggestion
	if !finished {
		r.Recorder.Event(
			database,
			corev1.EventTypeWarning,
			"Pending",
			fmt.Sprintf("Tenant creation operation is not completed, operationID: %s", operationID),
		)
		return Stop, ctrl.Result{RequeueAfter: DatabaseInitializationRequeueDelay}, nil
	}
```

Reply:
well, I have reversed the logic
<img width="321" alt="image" src="https://github.com/ydb-platform/ydb-kubernetes-operator/assets/1287547/3c00a2bd-e4dc-48f0-b9ab-705d617a0862">